spec/requests/topics_controller_spec.rb

```diff
@@ -2711,40 +2710,44 @@ def invite_group(topic, expected_status)
         # ugly, but no inteface to set this and we don't want to create
         # 100 posts to test this thing
         TopicView.stubs(:chunk_size).returns(2)
-      end
 
-      shared_examples "crawler layout" do |user_agent, via|
-        it "renders with the crawler layout, and handles proper pagination" do
-          get topic.url, env: { "HTTP_USER_AGENT" => user_agent, "HTTP_VIA" => via }
+        user_agent = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)"
 
-          body = response.body
+        get topic.url, env: { "HTTP_USER_AGENT" => user_agent }
 
-          expect(body).to have_tag(:body, with: { class: 'crawler' })
-          expect(body).to_not have_tag(:meta, with: { name: 'fragment' })
-          expect(body).to include('<link rel="next" href="' + topic.relative_url + "?page=2")
+        body = response.body
 
-          expect(response.headers['Last-Modified']).to eq(page1_time.httpdate)
+        expect(body).to have_tag(:body, with: { class: 'crawler' })
+        expect(body).to_not have_tag(:meta, with: { name: 'fragment' })
+        expect(body).to include('<link rel="next" href="' + topic.relative_url + "?page=2")
 
-          get topic.url + "?page=2", env: { "HTTP_USER_AGENT" => user_agent, "HTTP_VIA" => via }
-          body = response.body
+        expect(response.headers['Last-Modified']).to eq(page1_time.httpdate)
 
-          expect(response.headers['Last-Modified']).to eq(page2_time.httpdate)
+        get topic.url + "?page=2", env: { "HTTP_USER_AGENT" => user_agent}
+        body = response.body
+
+        expect(response.headers['Last-Modified']).to eq(page2_time.httpdate)
+
+        expect(body).to include('<link rel="prev" href="' + topic.relative_url)
```

Comment:
We have https://github.com/kucaahbe/rspec-html-matchers available in the test env so you might want to have a look at that.